src/App.js

```diff
@@ -14,11 +14,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const preparedGoods = goodsFromServer.map((item, index) => (
+  {
+    id: index,
+    name: item,
+  }
+));
+
+class App extends React.Component {
+  state = {
+    selectedGoods: [],
+  }
+
+toggleGood = ({ name }) => {
+  this.setState((state) => {
+    if (!state.selectedGoods.includes(name)) {
+      return {
+        selectedGoods: [...state.selectedGoods, name],
+      };
+    }
+
+    const goodIndex = state.selectedGoods.indexOf(name);
+
+    state.selectedGoods.splice(goodIndex, 1);
+
+    return {
+      selectedGoods: state.selectedGoods,
+    };
+  });
+}
+
+ clearSelect = () => {
+   this.setState({
+     selectedGoods: [],
+   });
+ }
+
+ render() {
+   const { selectedGoods } = this.state;
+
+   const temp = ((selectedGoods.length > 0) ? 1 : 0);
```

Comment:
don't you think it would be better to make it boolean? true/false === 1/0
```suggestion
   const temp = Boolean(selectedGoods.length);
```
and better to give it a more meaningful name. Like `areGoodsSelected` or smth like this

Reply:
I did't know it. Thank's.
 I'm going to use it in future.